lib/game.rb

```diff
@@ -0,0 +1,100 @@
+require_relative 'player'
+require_relative 'board'
+
+class Game
+
+  attr_reader :player_1, :player_2, :board, :current_player
+
+  def initialize
+    @selected_token = nil
+    @player_1 = welcome_player("player_1")
+    @player_2 = welcome_player("player_2")
+    @board = Board.new()
+    @current_player = @player_1
+  end
+
+  def play_turn
+    location = get_location()
+    board.update_field_values(current_player.token, location)
+    current_player.add_cell(location)
+  end
+
+  def populate_display
+    puts
+    puts "LEGEND:"
+    @board.draw_board(@board.legend_values)
+    puts "Use the legend to pick a space!"
+    puts
+    puts "GAME BOARD:"
+    @board.draw_board(@board.board_values)
+    puts
+  end
+
+  def switch_player
+    @current_player == @player_1  ? @current_player = @player_2 : @current_player = @player_1
+  end
+
+  def over?
+    win_conditions = [[1,2,3], [4,5,6], [7,8,9],
+                      [1,4,7], [2,5,8], [3,6,9],
+                      [1,5,9], [3,5,7]]
+
+    win_conditions.each do |condition|
+      if (condition - current_player.cells_chosen).empty?
+        puts "#{current_player.name} wins!"
+        return true
+      end
+    end
+    if board.board_values.none? { |value| value == " " }
+      puts "Tie game, no winners!"
+      return true
+    end
+    false
+  end
+
+  private
+
+  def welcome_player(player)
+    player_name = get_name(player)
+    player_token = get_token(player_name)
+    Player.new(player_name, player_token)
+  end
+
+  def get_name(player)
```

Comment:
Since instantiating a player should happen outside the Game class, perhaps this get_name method should be in the Player class.